test/e2e/common/util_mayastor_crds.go

```diff
@@ -318,3 +330,73 @@ func CheckForMSVs() (bool, error) {
 	}
 	return foundResources, err
 }
+
+func CheckAllMsvsAreHealthy() error {
+	msvGVR := schema.GroupVersionResource{
+		Group:    "openebs.io",
+		Version:  "v1alpha1",
+		Resource: "mayastorvolumes",
+	}
+
+	allHealthy := true
+	retrieveErrors := false
+	msvs, err := gTestEnv.DynamicClient.Resource(msvGVR).Namespace(NSMayastor).List(context.TODO(), metav1.ListOptions{})
+	if err == nil && msvs != nil && len(msvs.Items) != 0 {
+		for _, msv := range msvs.Items {
+			msvName, _ := retrieveFieldStringValue(&msv, "metadata", "name")
+			state, err := retrieveFieldStringValue(&msv, "status", "state")
+			if err == nil {
+				if state != "healthy" {
+					logf.Log.Info("CheckAllMsvsAreHealthy", "msvName", msvName, "state", state)
+					allHealthy = false
+				}
+			} else {
+				logf.Log.Info("CheckAllMsvsAreHealthy failed to access status.state", "msvName", msvName)
+				retrieveErrors = true
+			}
+		}
+	}
+
+	if retrieveErrors {
+		return fmt.Errorf("error accessing MSV status.state")
+	}
+	if !allHealthy {
+		return fmt.Errorf("all MSVs were not healthy")
+	}
+	return err
+}
+
+func CheckAllPoolsAreOnline() error {
+	msvGVR := schema.GroupVersionResource{
+		Group:    "openebs.io",
+		Version:  "v1alpha1",
+		Resource: "mayastorpools",
+	}
+
+	allHealthy := true
+	retrieveErrors := false
+	pools, err := gTestEnv.DynamicClient.Resource(msvGVR).Namespace(NSMayastor).List(context.TODO(), metav1.ListOptions{})
+	if err == nil && pools != nil && len(pools.Items) != 0 {
+		for _, pool := range pools.Items {
+			poolName, _ := retrieveFieldStringValue(&pool, "metadata", "name")
+			state, err := retrieveFieldStringValue(&pool, "status", "state")
+			if err == nil {
+				if state != "online" {
+					logf.Log.Info("CheckAllPoolsAreOnline", "pool", poolName, "state", state)
+					allHealthy = false
```

Comment:
Again, can't we just return here?

Reply:
Again, I wanted to test logs to show the complete set of pools that are not online.

Reply:
Ok.